sample_data_config/twitter-ads-api/twitter-ads-api.config.yaml

```diff
@@ -69,15 +69,48 @@ twitterAdsApi:
         parameterValues:
           fromBigQuery:
             projectName: 'elife-data-pipeline'
-            # query has been filtered for now because of api rate limit 
             sqlQuery: |-
-              SELECT DISTINCT t_data.id AS entity_id, CAST(CAST(t_data.created_at AS DATE) AS STRING) AS start_date
-              FROM `elife-data-pipeline.hc_dev.v_latest_twitter_campaign_details_elife_journal`
-              JOIN UNNEST(data) AS t_data
-              WHERE -- the campaigns before '2015-01-02' has to be filtered because of api limits
-              CAST(t_data.created_at AS DATE) > CAST('2015-01-02' AS DATE)
-              AND t_data.id IN ('i61cm')
-          maxPeriodInDays: 90
+              WITH t_all_campaigns AS(
+                SELECT 
+                  DISTINCT 
+                  t_data.id AS entity_id,
+                  CAST(t_data.created_at AS DATE) AS start_date
+                FROM `elife-data-pipeline.hc_dev.v_latest_twitter_campaign_details_elife_journal`
+                JOIN UNNEST(data) AS t_data
+                WHERE CAST(t_data.created_at AS DATE) > CAST('2015-01-02' AS DATE)
+              ),
+              t_max_start_time_by_campaign_and_placement AS (
+                SELECT 
+                  t_data.id AS entity_id,
+                  request.params.placement,
+                  MAX(request.params.start_time) OVER (PARTITION BY t_data.id, request.params.placement) AS max_start_time
+                FROM
+                `elife-data-pipeline.hc_dev.v_latest_twitter_campaign_clicks_and_impressions_elife_journal`
+                JOIN UNNEST(data) AS t_data
+              ),
+              t_min_start_time_per_campaign AS (
+                SELECT 
+                  entity_id,
+                  CAST(MIN(max_start_time) AS DATE) AS start_date,
+                FROM t_max_start_time_by_campaign_and_placement 
+                GROUP BY entity_id
+              ),
+              t_campaign_and_dates AS (
+                SELECT 
+                  t_campaing.entity_id,
+                  t_campaing.start_date AS campaign_start_date,
+                  IFNULL(t_calculated_date.start_date, t_campaing.start_date) AS start_date
+                FROM t_all_campaigns AS t_campaing
+                LEFT JOIN t_min_start_time_per_campaign AS t_calculated_date
+                ON t_campaing.entity_id = t_calculated_date.entity_id
+              )
+              SELECT 
+              entity_id, 
+              CAST(campaign_start_date AS STRING) AS campaign_start_date,
+              CAST(start_date AS STRING) AS start_date
+              FROM t_campaign_and_dates
+              WHERE ABS(DATE_DIFF(campaign_start_date, start_date, DAY)) < 91 -- (maxPeriodInDays - 7)
```

Comment:
Potential current test cases:

Assume maxPeriodInDays = 14

Campaign start date: 2022-01-01
Today (at this point): 2022-01-11

First run:

Start date: 2022-01-01 (campaign start date)

End date: 2022-01-08
Will fetch data for 2022-01-01 to 2022-01-07 (1 week)

Then:
Change current start date to: 2022-01-08
End date: 2022-01-10 (yesterday) (TODO check why not today)
Will fetch data for 2022-01-08 to 2022-01-09 (2 days)

Pipeline finishes

Next DAG run:
Today: 2022-01-12

New start date from BigQuery: 2022-01-08
End date: 2022-01-11 (yesterday)
Will fetch data for 2022-01-08 to 2022-01-10 (3 days)

...

Reply:
(TODO check why not today) --> checked now using today.